## PPOL564 Code discussion 2

### Rosey Cao
Sept. 10th 2020

### Preparation

In [1]:
## Read in the data (we would provide this)
import csv
with open('us-states.csv') as file:
        state_covid_data = []
        for row in csv.reader(file):
            state_covid_data.append(row)

print(len(state_covid_data))

10080


In [2]:
###Check the type and inspect the first 5 observations
type(state_covid_data)
state_covid_data[:5]

[['date', 'state', 'fips', 'cases', 'deaths'],
 ['2020-01-21', 'Washington', '53', '1', '0'],
 ['2020-01-22', 'Washington', '53', '1', '0'],
 ['2020-01-23', 'Washington', '53', '1', '0'],
 ['2020-01-24', 'Illinois', '17', '1', '0']]

### (1) Count up the number of _unique_ dates in the data. 

In [3]:
### Create a list of the data "column"
dates = []
for i in range(1, len(state_covid_data)):
    dates.append(state_covid_data[i][0])

### Create set of the dates to remove the duplicates; Calculate the length
print(len(set(dates)))

225

#### There are 225 unique dates in the data

### (2) Find the first date in which the District of Columbia recorded a case. 

In [4]:
### Create a list of the state "column"
state = []
for i in range(1, len(state_covid_data)):
    state.append(state_covid_data[i][1])

### Find the index of DC first case in the "state list"
DC_1stcase_index = state.index("District of Columbia")

### Use the index to find the date in the "state_covid_data"
print(state_covid_data[DC_1stcase_index][0])

'2020-03-07'

### (3) Write a function that takes in a _state name_ as input (e.g. "Wisconsin") and outputs the date of its first case.

In [5]:
### Create a function to locate date
def firstcase(StateStr):
    ##Create a list of the state "column"
    state = []
    for i in range(1, len(state_covid_data)):
        state.append(state_covid_data[i][1])

    ##Find the index of state's first case in the "statelist"
    StateStr_1stcase_index = state.index(StateStr)
    
    ##Use the index to find the date in the "state_covid_data"
    print(state_covid_data[StateStr_1stcase_index][0])

StateStr=input("Locate date of the first case. Please enter a state without quote marks!")
firstcase(StateStr)

Locate date of the first case. Please enter a state without quote marks!Wisconsin
2020-02-05


### Write a function that takes in a _state name_ as input (e.g. "Wisconsin") and outputs the date when the number of reported cases within the state exceeded 1000.

In [9]:
### Create a function to locate date
def exceed_1000(StateStr):
    ##Create a list of the state "column" with cases exceed 1000(skip those <1000)
    state_1000 = []
    for i in range(1, len(state_covid_data)):
        if eval(state_covid_data[i][3]) > 1000:
            state_1000.append(state_covid_data[i][1])
        else:
            continue
    print("Total number of items in the state list is {}".format(len(state_1000)))
    
    ##Create a list of the date "column" with cases exceed 1000(skip those <1000)
    date_1000 = []
    for i in range(1, len(state_covid_data)):
        if eval(state_covid_data[i][3]) > 1000:
            date_1000.append(state_covid_data[i][0])
        else:
            continue
    print("Total number of items in the date list is {}".format(len(date_1000)))

    ##Find the index of state's first case > 1000 within the filtered state list
    StateStr_1st1000_index = state_1000.index(StateStr)
    
    ##Use the index to locate the date within the filtered date list
    print(date_1000[StateStr_1st1000_index])
    
StateStr=input("Locate the date when cases within the state exceeded 1000.Please enter a state without quote marks!")
exceed_1000(StateStr)


Locate the date when cases within the state exceeded 1000.Please enter a state without quote marks!Florida
Total number of items in the state list is 7706
Total number of items in the date list is7706
2020-03-23


##### I added the number of items in the list just to make sure they are the same length. Life is much easier with  NumPy array and Pandas dataframe...But I'm trying to practice loops here!!